### DISCUSSION TITLE
Pnp search not working for some of the partial term

### DISCUSSION THREAD

**Discussion by supreet-bajaj** [microsoft-search/pnp-modern-search#3134] [General]
Hi, 

I am using PnP search box for showing one library results. But it’s not showing results if I am searching for some of the partial term. Like I have ‘picture34’ title of the Image in library when I am searching in library with term ‘Picture’ , no results are getting displayed but it’s working for some of the documents like term ‘test34’.

Could you please assist me what could be the issue.

Attached screenshots are for your reference.

![IMG_5668](https://github.com/microsoft-search/pnp-modern-search/assets/25117777/9c172f1c-7b23-46d5-96e4-208a1b1bf767)
![IMG_5669](https://github.com/microsoft-search/pnp-modern-search/assets/25117777/4d919d27-5c57-485a-8b25-adc1d30f4c36)
![IMG_5670](https://github.com/microsoft-search/pnp-modern-search/assets/25117777/215c7be2-8bec-4628-afd0-5f208de77403)

Please let me know if you have any questions.

Thanks,

**Comment by wobba** [comment]
You have to append * at the end to do substring search.  See https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#using-words-in-the-free-text-kql-query
